### PR TITLE
H801 LED Controller

### PR DESCRIPTION
**Pull request by erazor666**
Made a cookbook guide for the H801 LED controller.

### CODE REVIEW

File: esphomeyaml/cookbook/h801.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+H801 RGBW LED controller
+========================
+
+| The H801 is pretty afordable and easy to hack and adapt to your needs. It can be found on `Aliexpress <https://s.click.aliexpress.com/e/bbnUDBZW>`__
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/h801.rst]: Why the | in front?

File: esphomeyaml/cookbook/h801.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+| The H801 is pretty afordable and easy to hack and adapt to your needs. It can be found on `Aliexpress <https://s.click.aliexpress.com/e/bbnUDBZW>`__
+| `Here <http://tinkerman.cat/closer-look-h801-led-wifi-controller/>`__ is an article about the hardware for those who have an interest for more details about the board.
+| It can be used as a RGB, RGBW, RGB dual white or even control 5 individual monochromatic strips if you want or combinations of these.
+
+.. figure:: images/h801.jpg
+ :align: center
+ :width: 80.0%
+
+For esphomelib, you can then use the :doc:`RGBW </esphomeyaml/components/light/rgbw.html>`
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/h801.rst]: Do not add `.html` at the end of references.

File: esphomeyaml/cookbook/h801.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+.. figure:: images/h801.jpg
+ :align: center
+ :width: 80.0%
+
+For esphomelib, you can then use the :doc:`RGBW </esphomeyaml/components/light/rgbw.html>` 
+and the :doc:`ESP8266 Software PWM output </esphomeyaml/components/output/esp8266_pwm.html>` components using below configuration (the commented out code is for the RBG cold white strip in pictures):
+
+
+.. code-block:: yaml
+esphomeyaml:
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/h801.rst]: You have to leave a blank line after code-block and indent everything in here with four spaces

**Review by OttoWinter — COMMENTED** [review]
You'll also need to add an entry to `esphomeyaml/index.rst` together with a (small, jpg 8:10 aspect) thumbnail.

File: esphomeyaml/cookbook/h801.rst (hunk 1 above)
Comment (erazor666) [reply esphomeyaml/cookbook/h801.rst]: From reading some guidelines and making things stay on split up on lines like desired a guide said to use the | symbol in front, having some issues finding a good WYSIWG editor for RST and the docker image for esphomedocs didnt work for me :(

**Comment by erazor666** [comment]
There i think i got it all sorted.

**Comment by OttoWinter** [comment]
@erazor666 Uhm... no. The h801.rst seems to contain HTML now

**Comment by erazor666** [comment]
> @erazor666 Uhm... no. The h801.rst seems to contain HTML now

What the f... happend to that file? its been altered to some huge html document somehow? (not being the most experienced git user) but i've seen something like this happen if you try to "save as..." a file from a git repo sometimes.

File: esphomeyaml/cookbook/h801.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+For esphomelib, you can then use the :doc:`RGBW </esphomeyaml/components/light/rgbw>` 
+and the :doc:`ESP8266 Software PWM output </esphomeyaml/components/output/esp8266_pwm>` components using below configuration:
+
+
+.. code-block:: yaml
+    esphomeyaml:
+      name: h801test
+      platform: ESP8266
+      board: d1_mini
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/h801.rst:20]: There's not a d1_mini inside of this device. I think it is a `esp01_1m`. That would also explain why the OTA upload is not working (if you set the incorrect model OTA thinks the binary is too big and aborts)

File: esphomeyaml/cookbook/h801.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+This LED stip can be used with the sketch as-is
+
+.. figure:: images/rgbw.jpg
+ :align: center
+ :width: 80.0%
+
+For this type of led strip you have to swap the white and blue gpio numbers in the sketch
+
+
+
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/h801.rst:89]: Please remove these newlines (one empty line is enough)

File: esphomeyaml/cookbook/h801.rst [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+ :align: center
+ :width: 80.0%
+
+Back side of the board (don't melt the blue and red wire when soldering)
+
+.. figure:: images/rgbwarm.jpg
+ :align: center
+ :width: 80.0%
+
+This LED stip can be used with the sketch as-is
```
Comment (OttoWinter) [inline esphomeyaml/cookbook/h801.rst:80]: strip

**Review by OttoWinter — COMMENTED** [review]
Ok the guide looks quite good.

Unfortunately you'll have to rebase this PR on the `current` branch because all paths have changed a bit. Upside is with the change that caused that you'll also see instant deploy previews so you can update the PR here and you see the changes rendered within a minute.

Also, the images need to be resized. 600kb is quite big. Please downsize (+ compress, for example with https://tinyjpg.com/) all images. Especially the h801.jpg needs to be decreased a lot in size because it is on the main page (see other images in there for comparison; should be 8:10 aspect ratio with 512px max width

**Comment by erazor666** [comment]
Remade PR with the requested changes, still learning git. It's probably some better way to do this.

**Pull request closed without merging**